kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -140,9 +154,31 @@ public void writeRows(PartitionedTableId table,
         }
       }
     } while (retryCount <= retries);
-    throw new BigQueryConnectException(
-        String.format("Exceeded configured %d attempts for write request", retries),
-        mostRecentException);
+
+    if (mostRecentException == null) {
+      // indicate the most recent exception is partial write failure
+      throw new BigQueryConnectException(partialFailRowsMap);
```

Comment:
We still want to say that we exceeded our configured retry attempts.

Reply:
sure